crates/ruff/src/rules/flake8_comprehensions/fixes.rs

```diff
@@ -1312,5 +1312,8 @@ pub(crate) fn fix_unnecessary_comprehension_any_all(
     };
     tree.codegen(&mut state);
 
-    Ok(Edit::range_replacement(state.to_string(), expr.range()))
+    Ok(Fix::suggested(Edit::range_replacement(
```

Comment:
It's not entirely clear to me if the applicability should be decided here in `fixes.rs` or `rules/...`

Reply:
I think either is fine since the method is only used in `unnecessary_comprehension_any_all` (that's why I personally would move the function and co-locate it with the rule). 

Another reasoning: If we always know that a certain `Edit` is potentially unsafe, then it's okay to decide in the helper method.